kDrive/AppDelegate.swift

```diff
@@ -231,6 +243,56 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
         }
     }
 
+    func applicationDidBecomeActive(_ application: UIApplication) {
+        if let shortcutItem = shortcutItemToProcess {
+            @InjectService var accountManager: AccountManageable
+
+            guard let rootViewController = window?.rootViewController as? MainTabViewController else {
+                return
+            }
+
+            // Dismiss all view controllers presented
+            rootViewController.dismiss(animated: false)
+
+            guard let navController = rootViewController.selectedViewController as? UINavigationController,
+                  let viewController = navController.topViewController,
+                  let driveFileManager = accountManager.currentDriveFileManager else {
+                return
+            }
+
+            switch shortcutItem.type {
+            case Constants.applicationShortcutScan:
+                openScan(rootViewController, driveFileManager)
+            case Constants.applicationShortcutSearch:
+                let viewModel = SearchFilesViewModel(driveFileManager: driveFileManager)
+                viewController.present(SearchViewController.instantiateInNavigationController(viewModel: viewModel), animated: true)
+                MatomoUtils.track(eventWithCategory: .shortcuts, name: "search")
+            case Constants.applicationShortcutUpload:
+                openPhoto(rootViewController, driveFileManager)
+            case Constants.applicationShortcutSupport:
+                UIApplication.shared.open(URLConstants.support.url)
+                MatomoUtils.track(eventWithCategory: .shortcuts, name: "support")
+            default:
+                break
+            }
+
+            // reset the shortcut item
+            shortcutItemToProcess = nil
+        }
+    }
+
+    private func openScan(_ mainTabViewController: MainTabViewController, _ driveFileManager: DriveFileManager) {
```

Comment:
I think we can remove the `openScan` and `openPhoto` functions.
And put that code directly line 263 in the `switch shortcutItem.type {`

For now It should be OK, but if you find the function `applicationDidBecomeActive` too long. I guess we can add another file named `AppDelegate+Shortcuts.swift` to keep all the code related to shortcut handling, and keep the `openScan` and `openPhoto` functions there.

What do you think ?